components/server/src/workspace/gitpod-server-impl.ts

```diff
@@ -3001,24 +3001,31 @@ export class GitpodServerImpl implements GitpodServerWithTracing, Disposable {
         const user = await this.checkAndBlockUser("deleteTeam");
         traceAPIParams(ctx, { teamId, userId: user.id });
 
-        await this.guardTeamOperation(teamId, "delete", "not_implemented");
+        const { team, members } = await this.guardTeamOperation(teamId, "delete", "delete");
 
-        const teamProjects = await this.projectsService.getTeamProjects(teamId);
-        teamProjects.forEach((project) => {
+        const projects = await this.projectsService.getTeamProjects(teamId);
+
+        projects.forEach((project) => {
```

Comment:
Note I'm not yet removing relationships for projects. This will actually need to happen inside the `this.projectService.deleteProject`. There, we should also ensure we can do bulk delete by OrgID to simplify the looping we do here.

Reply:
And ideally, it would be transaction between the read and delete operations

Reply:
so now we need transactions that span multiple data access services